### PR TITLE
Fix geocoder logic

### PR DESCRIPTION
**Pull request by seanim0920**
* Add more geocoding test cases for places outside california
* Better error handling: Wrap geocoding lambda in an error catcher similar to the api lambda, and add more debug logs
* Print geocoder logs for failed test event creations, and change failed test event creation error message
  * Extracted the AWS logging function to a util
* Add region to the insert location sql
* Use short address value if the full address can't be formatted in geocoding function
* Ignore "Duplicate Entry" errors in the geocode handler, reducing the need for adding mock locations in tests.

Depends on tifshared pr https://github.com/tifapp/TiFShared/pull/93

https://trello.com/c/GqPfgwxS/864-geocoding-staging-bug-cant-create-events-with-placemark-more-location-tests-around-the-us

### CODE REVIEW

**Comment by tifapp** [comment]
Linked Ticket: https://trello.com/c/GqPfgwxS

File: GeocodingLambda/index.ts [hunk 1]
```diff
@@ -1,67 +1,9 @@
...
-  SearchCoordinatesForAddressAWS
-} from "./utils"
-
-// TODO: Fix handler type, fix util dependencies
+try {
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const { handler } = require("./handler")
+  exports.handler = handler
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-export const handler = (
```
Comment (tifapp) [inline GeocodingLambda/index.ts:20]: moved to handler.ts

File: GeocodingLambda/handler.test.ts [hunk 2]
```diff
@@ -1,153 +0,0 @@
-import { conn } from "TiFBackendUtils"
```
Comment (tifapp) [inline GeocodingLambda/handler.test.ts:1]: renamed integration.test.ts

File: GeocodingLambda/index.ts [hunk 3]
```diff
@@ -1,67 +1,8 @@
...
-                longitude
-              },
-              placemark
-            }
-          ))
-        })
-    )
-    .unwrap()
+} catch (error: any) {
+  console.error("Lambda error:", error.stack || error.message)
```
Comment (mhayes853) [inline GeocodingLambda/index.ts:6]: I think we could use a `Logger` instance here.

File: GeocodingLambda/integration.test.ts [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+    const [result1, result2, result3] =
+      await Promise.all([
+        handler(unknownAddressTestLocation.coordinate),
+        handler(unknownAddressTestLocation.coordinate),
+        handler(unknownAddressTestLocation.coordinate)
+      ])
+
+    expect(result1).toMatchObject(result2)
+    expect(result3).toMatchObject(result2)
+    expect(result1).toMatchObject(result3)
```
Comment (mhayes853) [inline GeocodingLambda/integration.test.ts]: It may be better to use an actual location here since `unknownAddressTestLocation` will always trigger an edge case.

File: GeocodingLambda/integration.test.ts [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+  })
+
+  afterAll(() => conn.closeConnection())
```
Comment (mhayes853) [inline GeocodingLambda/integration.test.ts:267]: Nit: We could make some shared test helper that autocloses the connection after all. Maybe this could just be a global jest setup that gets applied to all targets.

File: GeocodingLambda/utils.ts [hunk 6]
```diff
@@ -156,26 +158,36 @@ export const addLocationToDB = (
...
+        streetNumber,
+        postalCode,
+        region,
+        isoCountryCode
+      }
+    )
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  } catch (e: any) {
+    log.error("Could not insert location into cache: ", { error: e })
+    if (!(e.message.includes("Duplicate entry"))) {
```
Comment (mhayes853) [inline GeocodingLambda/utils.ts:185]: Nit: Would prefer that we checked an explicit error code instead of a string if possible.

File: GeocodingLambda/integration.test.ts [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+          coordinate: {
+            latitude: expect.closeTo(coordinate.value.latitude),
+            longitude: expect.closeTo(coordinate.value.longitude)
+          },
+          placemark: expect.objectContaining(
+            placemark.value
+          )
+        })
+
+        const cachedResult = await handler(coordinate, mockGeocoding, mockReverseGeocoding)
```
Comment (mhayes853) [inline GeocodingLambda/integration.test.ts:215]: Nit: For clarity, you could explicitly make the mock functions reject with a random error.

**Review by mhayes853 — COMMENTED** [review]
Thanks for replacing almost all of the `console.log` with calls to the logger.
